offgridplanner/optimization/helpers.py

```diff
@@ -271,3 +271,302 @@ def check_imported_demand_data(df, project_dict):
+                    "type": "object",
+                    "additionalProperties": {"type": "string"},
+                },
+                "custom_specification": {
+                    "type": "object",
+                    "additionalProperties": {"type": "string"},
+                },
+                "shs_options": {
+                    "type": "object",
+                    "additionalProperties": {"type": "integer"},
+                },
+                "consumer_detail": {
+                    "type": "object",
+                    "additionalProperties": {"type": "string"},
+                },
+                "is_connected": {
+                    "type": "object",
+                    "additionalProperties": {"type": "boolean"},
+                },
+            },
+            "additionalProperties": False,
+        },
+        "grid_design": {
+            "type": "object",
+            "required": ["distribution_cable", "connection_cable", "pole", "mg", "shs"],
+            "properties": {
+                "distribution_cable": {
+                    "type": "object",
+                    "required": ["capex", "max_length", "epc"],
+                    "properties": {
+                        "capex": {"type": "number"},
+                        "max_length": {"type": "number"},
+                        "epc": {"type": "number"},
+                    },
+                },
+                "connection_cable": {
+                    "type": "object",
+                    "required": ["capex", "max_length", "epc"],
+                    "properties": {
+                        "capex": {"type": "number"},
+                        "max_length": {"type": "number"},
+                        "epc": {"type": "number"},
+                    },
+                },
+                "pole": {
+                    "type": "object",
+                    "required": ["capex", "max_n_connections", "epc"],
+                    "properties": {
+                        "capex": {"type": "number"},
+                        "max_n_connections": {"type": "integer"},
+                        "epc": {"type": "number"},
+                    },
+                },
+                "mg": {
+                    "type": "object",
+                    "required": ["connection_cost", "epc"],
+                    "properties": {
+                        "connection_cost": {"type": "number"},
+                        "epc": {"type": "number"},
+                    },
+                },
+                "shs": {
+                    "type": "object",
+                    "required": ["include", "max_grid_cost"],
+                    "properties": {
+                        "include": {"type": "boolean"},
+                        "max_grid_cost": {"type": "number"},
+                    },
+                },
+            },
+        },
+        "yearly_demand": {"type": "number"},
+    },
+    "additionalProperties": False,
+}
+
+GRID_V2_SCHEMA = {
```

Comment:
This is the jsonschema for the structure implemented by @CelinaKellinghaus. The problem I see for validation is that the items in each "nodes" items could be swapped and we would only notice later in the code. Like those ones:
``` 
                    {"type": "string"},  # how_added
                    {"type": "string"},  # node_type
                    {"type": "string"},  # consumer_type
```

This is only a draft jsonschema, one could be more restrictive with the allowed values. I think we should favor a format which reduces the size of the transaction.